src/main/java/com/adaptavist/tm4j/jenkins/cucumber/CucumberReportParser.java

```diff
@@ -0,0 +1,182 @@
+package com.adaptavist.tm4j.jenkins.cucumber;
+
+import com.google.gson.stream.JsonReader;
+import com.google.gson.stream.JsonToken;
+import com.google.gson.stream.JsonWriter;
+
+import java.io.IOException;
+
+// this parser was based on https://github.com/cucumber/cucumber-json-schema/blob/main/schema.json
+public class CucumberReportParser {
```

Comment:
I was expecting this code to only remove the `embedded` property but honestly, considering that the cucumber json schema is correct, this is a good idea as it's more strict and can prevent some other issues as well. My only concern is that, in case the schema changes or there is a new schema available (new format, new attributes), we are in trouble, this "strict mode" approach will become a problem. 

Apart from that, i remember the C4J guys mentioning something like "certain tools generate some slightly different cucumber file result formats", something similar that we faced with xUnit file, they might help with this question.

Have you considered some possible schema change or/and result file differences? Do you think it's important to consider?

Reply:
The file from the ticket has different format from cucumber file, but it have the basic fields from the cucumber schema.   This class will only filter the fields that it doesn’t , and send to cucumber only the fields from the schema. 

Yes, this would fail if the file doesn't have the `cucumber-json-schema` but I think that cucumber from jira would fail too.

If cucumber change the schema , yes we would need to change this class.

Reply:
Yeah, I thought the same thing... That perhaps it'd be better to have a block-list of properties instead of an allow-list. I have mixed feelings about it but I think we can go on with the allow-list approach and see if we run into any issues that a block-list would prevent.